src/main/java/edu/uwaterloo/ece658/service/S3Service.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package edu.uwaterloo.ece658.service;
+import java.net.URL;
+
+import javax.ejb.Stateless;
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.HttpMethod;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.DeleteObjectRequest;
+import com.amazonaws.services.s3.model.GeneratePresignedUrlRequest;
+
+/**
+ *
+ * @author Daniel
+ */
+@Stateless
+public class S3Service {
+
+    // TODO: assign IAM to each user for S3 access
+    final private String accessKey = "AKIAI3V5NZVUSNCMDN7Q";
+    final private String secretKey = "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";
+    final private String bucketName = "phototrest";
+    
+    public URL getUploadURL(String s3Key) {
+        URL uploadURL = getS3Url(s3Key, HttpMethod.PUT);
+        return uploadURL;
+    }
+    
+    public URL getDownloadURL(String S3key) {
+        URL downloadURL = getS3Url(S3key, HttpMethod.GET);
+        return downloadURL;
+    }
+    
+    /**
+     * Returns the URL for a specified key. Specify method for upload or download
+     * @param objectKey 
+     * @param method (PUT = upload, GET = download)
+     * @return 
+     */
+    private URL getS3Url(String objectKey, HttpMethod method) {
+        BasicAWSCredentials creds =
+                new BasicAWSCredentials(accessKey, secretKey);
+        
+        AmazonS3 s3Client = AmazonS3ClientBuilder.standard().withCredentials(

```

Comment:
Can you make s3Client a field, so that you don't need to create it again for efficiency?